common4j/src/main/com/microsoft/identity/common/java/telemetry/Telemetry.java

```diff
@@ -280,6 +280,38 @@ public void flush(@NonNull final String correlationId) {
         }
     }
 
+    /**
+     * Get the final Telemetry Map for a correlationId.
+     * @param correlationId The correlation id should either passed by the client app through the API call
+     *                      or generated by the API dispatcher.
+     */
+    public List<Map<String, String>> getFinalMap(@NonNull final String correlationId) {
+        if (!mIsTelemetryEnabled) {
+            return Collections.emptyList();
+        }
+
+        if (StringUtil.isNullOrEmpty(correlationId)) {
+            Logger.warn(TAG, "No correlation id set.");
+            return Collections.emptyList();
+        }
+
+        //check the configuration
+        if (!mDefaultConfiguration.isDebugEnabled() && mIsDebugging) {
+            return Collections.emptyList();
+        }
+
+        List<Map<String, String>> finalRawMap = new CopyOnWriteArrayList<>();
```

Comment:
```suggestion
        final List<Map<String, String>> finalRawMap = new CopyOnWriteArrayList<>();
```

Reply:
Why this data structure?

Reply:
Unless there's some reason to require a CopyOnWriteArrayList, I'd suggest just using a normal ArrayList.